rplidar.go

```diff
@@ -348,40 +382,50 @@ func (d *Device) scan(options lidar.ScanOptions) (lidar.Measurements, error) {
 
 			nodeAngle := (float64(node.GetAngle_z_q14()) * 90 / (1 << 14))
 			nodeDistance := float64(node.GetDist_mm_q2()) / 4
-			measurements = append(measurements, lidar.NewMeasurement(nodeAngle, nodeDistance/1000))
+
+			err := pc.Set(pointFrom(rdkUtils.DegToRad(nodeAngle), rdkUtils.DegToRad(0), float64(nodeDistance)/1000, 255))
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	if len(measurements) == 0 {
+	if pc.Size() == 0 {
 		return nil, nil
 	}
-	if options.NoFilter {
-		return measurements, nil
+
+	t_now := (time.Now())
+	t_now2 := t_now.Format(time.RFC3339)
+	t_str := strings.Replace(strings.Replace(t_now2, ":", "_", 3), "-", "", 2)
+
+	f, err := os.Create("data/rplidar_data_" + t_str + ".pcd")
+	if err != nil {
+		return nil, err
 	}
-	sort.Stable(measurements)
-	filteredMeasurements := make(lidar.Measurements, 0, len(measurements))
-
-	minAngleDiff, maxDistDiff := d.filterParams()
-	prev := measurements[0]
-	detectedRay := false
-	for mIdx := 1; mIdx < len(measurements); mIdx++ {
-		curr := measurements[mIdx]
-		currAngle := curr.AngleDeg()
-		prevAngle := prev.AngleDeg()
-		currDist := curr.Distance()
-		prevDist := prev.Distance()
-		if math.Abs(currAngle-prevAngle) < minAngleDiff {
-			if math.Abs(currDist-prevDist) > maxDistDiff {
-				detectedRay = true
-				continue
-			}
-		}
-		prev = curr
-		if !detectedRay {
-			filteredMeasurements = append(filteredMeasurements, curr)
-		}
-		detectedRay = false
+	defer utils.UncheckedErrorFunc(f.Close)
```

Comment:
So the notes on this UncheckedErrorFunc (in goutils) explcitly say "never use this for closing writers" and "f" is is definitely a file writer.

https://github.com/viamrobotics/goutils/blob/ee8bd5e1fc8bdc0e3449e936988a6d2f3d7da508/error.go#L72

This close should be handled explicitly and checked for errors.

Reply:
I stumbled also over that comment - but thought that `f` is a file, whereas `w` is a writer. Am I mistaken?

Reply:
In Go parlance, anything that supports the "io.Writer" interface is a "writer." That definitely includes files, but a lot of other stuff too. Files are a particularly bad case though, because Write() isn't always guaranteed to be on disk until after the file is closed. So if you did a bunch of writes (with no error) the Close() itself can return an error about "can't write to disk" or similar, and even the previous Write() opertions may be lost.

If you're curious, a much more in depth explanation is here: https://www.joeshaw.org/dont-defer-close-on-writable-files/

Reply:
Thanks for explaining and for the link!

Done.